Update and expose ZeroRedundancyOptimizer docs

mrshenli opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#52937 Update and expose ZeroRedundancyOptimizer docs**

Differential Revision: [D26696938](https://our.internmc.facebook.com/intern/diff/D26696938)

mrshenli committed:
Update and expose ZeroRedundancyOptimizer docs

[ghstack-poisoned]

mrshenli commented:
![screencapture-file-Users-shenli-tmp-html-distributed-optim-html-2021-02-26-14_24_04](https://user-images.githubusercontent.com/16999635/109345586-6273e500-783e-11eb-9c78-0e5dd14a73d1.png)

mrshenli committed:
Update on "Update and expose ZeroRedundancyOptimizer docs"


Differential Revision: [D26696938](https://our.internmc.facebook.com/intern/diff/D26696938)

[ghstack-poisoned]

blefaudeux reviewed (approved):
LGTM, thanks @mrshenli

mrshenli committed:
Update on "Update and expose ZeroRedundancyOptimizer docs"


Differential Revision: [D26696938](https://our.internmc.facebook.com/intern/diff/D26696938)

[ghstack-poisoned]

mrshenli committed:
Update on "Update and expose ZeroRedundancyOptimizer docs"


Differential Revision: [D26696938](https://our.internmc.facebook.com/intern/diff/D26696938)

[ghstack-poisoned]

mrshenli commented:
this breaks master, reverting
